### PR TITLE
More documentation

### PR DESCRIPTION
**Pull request by smondet**
Of course there will be more, but this makes already a reasonably sized PR.

I've pushed there for now: http://wr.mondet.org/smondet/biokepi-doc/

---
This change is [Reviewable](https://reviewable.io/reviews/hammerlab/biokepi/242)

### CODE REVIEW

File: src/lib/biokepi.ml [hunk 1]
```diff
@@ -1,45 +1,212 @@
...
+*)
+
+(** The Embedded Bioinformatics Domain Specific Language
+
+     This Embedded DSL is implemented following the “Typed Tagless Final
+     Interpreter” method.
+
+    It's usage is as follows:
+
+    - Write EDSL expressions inside a functor taking {!Biokepi.EDSL.Semantics}
```
Comment (ihodes) [inline src/lib/biokepi.ml]: might be nice to...ahem...describe the meaning of Biokepi.EDSL.Semantics

File: src/lib/biokepi.ml [hunk 2]
```diff
@@ -1,45 +1,212 @@
...
+   Top-level entry point into the library
+
+
+   Biokepi provides different levels of abstraction to construct (Ketrew)
+   pipelines:
+
+   - {!Biokepi.EDSL}: very high-level embedded domain specific language that
+   encode very consise complex pipelines and uses OCaml type-system to verify
+   properties on them. The EDSL expressions can be “compiled” to JSON, to
+   Graphviz ["dot"] graphs, … {!Biokepi.Pipeline} is an earlier, soon to be
```
Comment (ihodes) [inline src/lib/biokepi.ml]: Most importantly, to a workflow that can be submitted to the Ketrew server

File: src/doc/Guide_to_the_code.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
+For Developpers
+===============
+
+
+Libraries
+---------
+
+Biokepi is made of 5 libraries: 4 building blocs and wrapping module
```
Comment (ihodes) [inline src/doc/Guide_to_the_code.md]: and 1 wrapping module*

File: src/doc/Guide_to_the_code.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+(`Biokepi`).
+
+`Biokepi` contains a single module `./src/lib/biokepi.ml` that wraps modules
+form the other 4 libraries under a cleaner API.
```
Comment (ihodes) [inline src/doc/Guide_to_the_code.md]: form -> from

**Comment by ihodes** [comment]
LGTM, few comments. Also, the types on this page are totally unreadable: http://wr.mondet.org/smondet/biokepi-doc/Module_Hierarchy.html

**Comment by smondet** [comment]
@ihodes I've made the SVGs `img`s also links to the actual SVG so you can click to open them and use the zoom of your browser. But yeah still pretty tiny :)

**Comment by ihodes** [comment]
Awesome, thank you! LGTM

**Pull request merged**